src/libraries/System.Private.CoreLib/src/System/Globalization/JapaneseCalendar.cs

```diff
@@ -66,18 +66,25 @@ public partial class JapaneseCalendar : Calendar
         //     english name, and abbreviated english names.
         internal static EraInfo[] GetEraInfo()
         {
-            // See if we need to build it
+            if (s_japaneseEraInfo == null)
+            {
+#if TARGET_MACCATALYST || TARGET_IOS || TARGET_TVOS
+                s_japaneseEraInfo = GlobalizationMode.UseNls ? NlsGetJapaneseEras() : (GlobalizationMode.Hybrid ? GetJapaneseErasNative() : IcuGetJapaneseEras());
+#else
+                s_japaneseEraInfo = GlobalizationMode.UseNls ? NlsGetJapaneseEras() : IcuGetJapaneseEras();
+#endif
+            }
+
             return s_japaneseEraInfo ??
-                (s_japaneseEraInfo = GlobalizationMode.UseNls ? NlsGetJapaneseEras() : IcuGetJapaneseEras()) ??
                 // See if we have to use the built-in eras
-                (s_japaneseEraInfo = new EraInfo[]
+                new EraInfo[]
```

Comment:
Why are we no longer setting `s_japaneseEraInfo` if it is found to be `null` here?

Reply:
good catch, will update.